stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Attachment.kt

```diff
@@ -41,7 +41,7 @@ public data class Attachment(
 ) : CustomObject {
 
     public sealed class UploadState {
-        public object InProgress : UploadState()
+        public data class InProgress(val currentProgress: Long) : UploadState()
```

Comment:
`currentProgress` here is a percentage value from 0 to 100, right? We should probably document that.

Also, did we give up on adding two parameters here, a current bytes + a total bytes value?

Reply:
Yes, we need to document the progress but I am little unsure about if we keep percentage here or two parameters, `bytesRead`, `totalBytes`.  I ditched the two parameters idea for the moment as ProgressCallback is public class and we will have to change the arguments of `onProgress` method also. That will be a breaking change. What do you suggest here?

Reply:
Perhaps we could add a new method in `ProgressCallback` with two parameters, give it a default empty implementation. We could deprecate the old one, telling people to implement the new one instead.

Perhaps we could have the default implementation call back into the old one as its default implementation.